### DISCUSSION TITLE
[New Feature] TeamTypes + Script Interruption Mechanic

### DISCUSSION THREAD

**Discussion by ErastusMercy** [Vinifera-Developers/Vinifera#1412] [Feature Requests]
### Description:
Teams in Tiberian Sun are required to finish their script before being recruited to a new team and being given new orders. This can make units behave strangely if their script would ever need to be interrupted by events like player actions making their current orders unnecessary. A way to allow a script to be interrupted would provide much greater control over Teams and Team Recruitment. This will likely also affect AI Triggers and could be applied there as well.

### Possible Implementation:
A 'Script command' which checks for Local and/or Global variables. If the variable is "set", it either:

- Skips the next line (or any other line) in the script, thus allowing the script to Fork between two different outcomes.
- Completes the full script action, thus allowing the unit to be recruited into a new Team.

**Comment by Crimsonum** [comment]
A script can be interrupted either by having a trigger destroy the team which leaves the team members open for recruiting into a new team, or by making another team recruit the team members directly. A team with higher `Priority` can recruit team members from another team with lower `Priority` value. `Group` should also match between these teams unless `Recruiter` is used on the recruiting team.

Thus it seems like this issue is covered, unless there's an edge case or something where these methods don't work?
